src/mem/Allocator.h

```diff
@@ -29,22 +36,33 @@ class Allocator : public Interface
      */
     virtual size_t getStats(PoolStats &) = 0;
 
-    virtual PoolMeter const &getMeter() const = 0;
+    /// get current statistics tracked for this allocator
+    virtual PoolMeter const &getMeter() const { return meter; }
+    virtual PoolMeter &getMeter() { return meter; }
```

Comment:
I think we should just make `meter` public because nobody overrides this virtual method, and I doubt we will need to override it in the foreseeable future. The proposed design is also dangerous -- it would be trivial for some future Allocator code to access `meter` instead of a very different meter accessible via getMeter()! Keeping the `meter` data member public will also reduce the footprint of this PR and improve performance of metering code.

If you are sure getMeter() might become overwritten in the future, then just comment out the `virtual` keyword for now.

Reply:
That is polish which I am avoiding for this PR.

Reply:
This PR introduces a new virtual method and makes the "virtual" in the old const method unnecessary. This PR adds a new data member and makes that data member private (necessitating adding a non-const method and creating a conflict for the Allocator code -- which meter access avenue do I use?). Addressing these significant problems _created_ by this PR is clearly in this PR scope, and it is not "polish" that should be avoided in this PR, especially since there exist a trivial way to address these problems: make the new data member public (with the name matching the old one), remove the word "virtual" from the old const method, and not add the non-const method.

Reply:
see https://github.com/squid-cache/squid/blob/master/src/mem/Pool.h#L139
and https://github.com/squid-cache/squid/blob/master/src/mem/Pool.h#L146

All problems you are stating as new exist in the MemImplementingAllocator current official API. This PR merged the two APIs

The 'meter' member was previously protected (not public) and existence of both accessors made that dangerous. A non-const pool child could access meter directly and alter it during a const method.

Reply:
> All problems you are stating as new exist in the MemImplementingAllocator current official API. This PR merged the two APIs

In a hypothetical scenario where a PR mechanically merges two classes without any other changes, the above argument would have been valid[^1]. This PR does not mechanically merge two classes without any other changes. This PR adds new types, renames fields, changes fields visibility, documents controversial APIs, etc. Those changes, once polished/adjusted, are the _right_ thing to do in principle -- the merger should improve code quality rather than make it worse, of course. However, the presence of those (good) changes invalidates "mechanical mergers should avoid API fixes" arguments as inapplicable. We are dealing with a careful manual elimination of an unwanted hierarchy member, not a mechanical merger of two classes.

> The 'meter' member was previously protected (not public) and existence of both accessors made that dangerous.

The old code was problematic. There is no need to preserve/move some of the old problems in/into Allocator. Implementing the trivial changes suggested in this PR will avoid/fix problems instead of preserving them (in some areas) and making them worse (in others).

> A non-const pool child could access meter directly and alter it during a const method.

A const method cannot alter a regular data member, regardless of that member visibility or originating class.

In general:

* A public data member is less dangerous than a combination of a private data member and a virtual accessor.
* A public data member is as dangerous as a public non-virtual modifier method.
* Non-virtual methods are better than virtual methods that are never overridden.

The trivial changes requested here will make the code better/safer. We should not waste time on finding reasons to avoid them.

[^1]: That argument would still not be enough to justify creating bad merged API in most cases (including this one), but at least it would have been _valid_ on the same "mechanical" level.

Reply:
FTR, IMO exposing meter is useful to reduce virtual table size. The other arguments being presented do not make much sense.
To avoid arguing, done in 64d513cf92105fd06905964ae0177f57a6baf675

Reply:
In the future, if you want to avoid arguing, please avoid posting your own arguments (_any_ arguments and especially arguments that can be misinterpreted as an attack on others!) just before implying that no followup is wanted. Adding that "to avoid arguing" phrase is enough to both imply that no followup is wanted and to record one's disagreement with previous arguments.